fix for shutdown up from vdr-1.5.x

Joerg Bornkessel <hd_brummy@gentoo.org> 25 Mar 2007
diff -Naur setup-0.3.1.orig/setupmenu.cpp setup-0.3.1/setupmenu.cpp
--- setup-0.3.1.orig/setupmenu.cpp	2007-03-25 22:02:45.214194576 +0200
+++ setup-0.3.1/setupmenu.cpp	2007-03-25 22:09:43.522111889 +0200
@@ -11,6 +11,9 @@
 
 #include <stdlib.h>
 #include <vdr/menuitems.h>
+#if APIVERSNUM > 10500
+#include <vdr/shutdown.h>
+#endif
 #include <string>
 
 #include "setupmenu.h"
@@ -856,7 +859,11 @@
             case kYellow:
                         if (Interface->Confirm(cRecordControls::Active() ? tr("Recording - restart anyway?") : tr("Really restart?")))
                         {
+#if APIVERSNUM > 10500
+						  ShutdownHandler.RequestEmergencyExit();
+#else
                           cThread::EmergencyExit(true);
+#endif
                           return osEnd;
                         }
                         return osContinue;
@@ -864,7 +871,11 @@
             case kBlue:
                         if (Interface->Confirm(cRecordControls::Active() ? tr("Recording - restart anyway?") : tr("Really reboot?")))
                         {
+#if APIVERSNUM > 10500
+						  ShutdownHandler.RequestEmergencyExit();
+#else
                           cThread::EmergencyExit(true);
+#endif
                           system(_config->GetBootLinux());
                         }
                         return osContinue;
